crates/cxx-qt-gen/src/lib.rs

```diff
@@ -214,4 +249,46 @@ mod tests {
         let expected_output = format_rs_source(include_str!("../test_outputs/signals.rs"));
         assert_str_eq!(rust, expected_output);
     }
+
+    #[test]
+    fn generates_types_primitive_property_rust() {
+        let parser = Parser::from(
+            syn::parse_str(include_str!("../test_inputs/types_primitive_property.rs")).unwrap(),
+        )
+        .unwrap();
+        let generated = GeneratedRustBlocks::from(&parser).unwrap();
+        let rust = format_rs_source(&write_rust(&generated).to_string());
+
+        let expected_output =
+            format_rs_source(include_str!("../test_outputs/types_primitive_property.rs"));
+        assert_str_eq!(rust, expected_output);
+    }
+
+    #[test]
+    fn generates_types_qt_property_rust() {
+        let parser = Parser::from(
+            syn::parse_str(include_str!("../test_inputs/types_qt_property.rs")).unwrap(),
+        )
+        .unwrap();
+        let generated = GeneratedRustBlocks::from(&parser).unwrap();
+        let rust = format_rs_source(&write_rust(&generated).to_string());
+
+        let expected_output =
+            format_rs_source(include_str!("../test_outputs/types_qt_property.rs"));
+        assert_str_eq!(rust, expected_output);
+    }
+
+    #[test]
+    fn generates_types_qt_invokable_rust() {
+        let parser = Parser::from(
+            syn::parse_str(include_str!("../test_inputs/types_qt_invokable.rs")).unwrap(),
+        )
+        .unwrap();
+        let generated = GeneratedRustBlocks::from(&parser).unwrap();
+        let rust = format_rs_source(&write_rust(&generated).to_string());
+
+        let expected_output =
+            format_rs_source(include_str!("../test_outputs/types_qt_invokable.rs"));
+        assert_str_eq!(rust, expected_output);
+    }
```

Comment:
This test code is redundant. It could be refactored into a common function that takes the filename as an argument.

Reply:
Done